src/hnsw.c

```diff
@@ -195,6 +209,180 @@ static float8 vector_dist(Vector *a, Vector *b, usearch_metric_kind_t metric_kin
+    }
+
+    // Check if table Oid is invalid
+    if(table_id == InvalidOid) {
+        ereport(ERROR, (errmsg("Table %s not found in schema %s", tablename_str, schema_name)));
+    }
+
+    // Open the heap relation
+    Relation heapRelation = table_open(table_id, AccessShareLock);
+    Oid      relation_oid = RelationGetRelid(heapRelation);
+    elog(INFO, "Relation OID: %u", relation_oid);
+
+    // TODO: Check if table is unlogged (supported at the moment)
+    // if(!RelationIsLogicallyLogged(heapRelation)) {
+    //     ereport(ERROR,
+    //             (errmsg("Table %s is unlogged. HNSW index on unlogged tables is not supported.",
+    // tablename_str)));
+    // }
+
+    // Create indexRelationName based on table name and column name
+    char *indexRelationName = palloc(strlen(table_name) + strlen("_hnsw_idx") + 1);
+    snprintf(indexRelationName, strlen(table_name) + strlen("_hnsw_idx") + 1, "%s_hnsw_idx", table_name);
+
+    // Create single node list for column name
+    List *indexColNames = list_make1("vector");
+
+    // Get column Oid
+    Oid   colId = getColumnOID(table_name, "vector");
+    List *ii_Expressions = list_make1_oid(colId);
+    elog(INFO, "colId: %u", colId);
+
+    // Get the access method OID for HNSW
+    Oid accessMethodObjectId = get_am_oid("hnsw", false);
+
+    // Set up the IndexInfo structure
+    IndexInfo *indexInfo = makeNode(IndexInfo);
+    indexInfo->ii_NumIndexAttrs = 1;
+    indexInfo->ii_NumIndexKeyAttrs = 1;
+    indexInfo->ii_Expressions = NIL;
+    indexInfo->ii_ExpressionsState = NIL;
+    indexInfo->ii_Predicate = NIL;
+    indexInfo->ii_PredicateState = NULL;
+    indexInfo->ii_ExclusionOps = NULL;
+    indexInfo->ii_ExclusionProcs = NULL;
+    indexInfo->ii_ExclusionStrats = NULL;
+    indexInfo->ii_OpclassOptions = NULL;
+    indexInfo->ii_Unique = false;
+#if PG_VERSION_NUM >= 150000
+    indexInfo->ii_NullsNotDistinct = false;
+    indexInfo->ii_ReadyForInserts = false;
+    indexInfo->ii_CheckedUnchanged = false;
+    indexInfo->ii_IndexUnchanged = false;
+#endif
+    indexInfo->ii_Concurrent = false;
+    indexInfo->ii_BrokenHotChain = false;
+    indexInfo->ii_ParallelWorkers = 0;
+    indexInfo->ii_Am = accessMethodObjectId;
+    indexInfo->ii_AmCache = NULL;
+    indexInfo->ii_Context = CurrentMemoryContext;
+
+    // Set the access method for the index
+    indexInfo->ii_Am = accessMethodObjectId;
+
+    elog(INFO, "accessMethodObjectId: %u", accessMethodObjectId);
+
+    // Get tableSpaceId
+    bool partitioned = heapRelation->rd_rel->relkind == RELKIND_PARTITIONED_TABLE;
+    Oid  tableSpaceId = get_rel_tablespace(table_id);
+    if(tableSpaceId == InvalidOid) {
+        tableSpaceId = GetDefaultTablespace(heapRelation->rd_rel->relpersistence, partitioned);
+    }
+
+    elog(INFO, "tableSpaceId: %u", tableSpaceId);
+
+    // Set the Oids for collations and ops
+    Oid   collationObjectId[] = {InvalidOid};
+    Oid   classObjectId[] = {InvalidOid};
```

Comment:
@Ngalstyan4 I have a question on this as well, not sure what goes here. I have tried to understand what these do to no avail